client/jetpack-connect/help-button.jsx

```diff
@@ -15,7 +15,7 @@ import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
 import { localize } from 'i18n-calypso';
 import { recordTracksEvent } from 'state/analytics/actions';
 
-export class JetpackConnectHelpButton extends PureComponent {
+export class JetpackConnectHelpButton extends Component {
```

Comment:
I'm reluctant to accept this change - I really don't think changing from pure components to components solves the problem in the long term.

This could easily be a problem for many other locations throughout Calypso - should we fix it that way there as well? I don't think so - this only conceals the underlying problem.

I think that we should find a solution for the core problem, instead of requiring us to go with impure solutions that could potentially cause performance bottlenecks at some point (even if that's not the case right now).

Reply:
Thanks for the comments @tyxla !

> I think that we should find a solution for the core problem

I agree with this 💯%. I'm going to look into this using you're already-mentioned feedback. The main issue was presenting the user with a fully-translated NUX flow, which, at the time, I considered a higher priority than pre-optimizing components. 

> This could easily be a problem for many other locations throughout Calypso - should we fix it that way there as well?

Also agree that the answer is 'no'. Framework considerations are important of course, yet it's a pretty bad experience to push a user, especially one who has decided to sign up with Jetpack for the first time, from their language to `en` without warning. 

Nevertheless, you just reminded me that we're in a marathon :)  So I'll put this PR on hold while we I a quick look at `i18n.localize`.  If we can get something through I'll update this PR accordingly since I'd still like to get the route changes through. 

Cheers again!

Reply:
Would appreciate a review! https://github.com/Automattic/i18n-calypso/pull/77 
🙇

Reply:
Woot, awesome, I appreciate your diligence @ramonjd, thanks!